x/evm/keeper/abci.go

```diff
@@ -44,5 +52,33 @@ func (k *Keeper) EndBlock(ctx context.Context) error {
 	bloom := ethtypes.BytesToBloom(k.GetBlockBloomTransient(infCtx).Bytes())
 	k.EmitBlockBloomEvent(infCtx, bloom)
 
+	feeCollectorAddr := authtypes.NewModuleAddress(authtypes.FeeCollectorName)
+	balance := k.bankKeeper.GetBalance(sdkCtx, feeCollectorAddr, k.GetParams(sdkCtx).EvmDenom)
+	if balance.IsZero() {
+		return nil
```

Comment:
Let's add info logs for all corner cases where we abandon the transfer process. This is an important logic and its observability is crucial.

Reply:
[99cf088](https://github.com/mezo-org/mezod/pull/417/commits/99cf0889e10a8f96528b92ef8e15f4ff30644a43)